autopush/web/registration.py

```diff
@@ -145,45 +149,43 @@ def post(self, *args, **kwargs):
 
 
         """
-        self.start_time = time.time()
         self.add_header("Content-Type", "application/json")
-        params = self.valid_input['params']
+
+        uaid = self.valid_input['uaid']
+        router = self.valid_input["router"]
+        router_type = self.valid_input["router_type"]
+        router_token = self.valid_input.get("router_token")
+        router_data = self.valid_input['router_data']
+
         # If the client didn't provide a CHID, make one up.
         # Note, valid_input may explicitly set "chid" to None
         # THIS VALUE MUST MATCH WHAT'S SPECIFIED IN THE BRIDGE CONNECTIONS.
         # currently hex formatted.
-        self.chid = params["channelID"] = (self.valid_input["chid"] or
+        chid = router_data["channelID"] = (self.valid_input["chid"] or
                                            uuid.uuid4().hex)
         self.ap_settings.metrics.increment("updates.client.register",
                                            tags=self.base_tags())
-        # If there's a UAID, ensure its valid, otherwise we ensure the hash
-        # matches up
-        new_uaid = False
-
-        # normalize the path vars into parameters
-        router = self.ap_settings.routers[self.valid_input['router_type']]
-
-        if not self.valid_input['uaid']:
-            self.valid_input['uaid'] = uuid.uuid4()
-            new_uaid = True
-        self.uaid = self.valid_input['uaid']
-        self.app_server_key = params.get("key")
-        if new_uaid:
-            d = Deferred()
-            d.addCallback(router.register, router_data=params,
-                          app_id=self.valid_input.get("router_token"),
-                          uri=self.request.uri)
-            d.addCallback(self._save_router_data,
-                          self.valid_input["router_type"])
-            d.addCallback(self._create_endpoint)
-            d.addCallback(self._return_endpoint, new_uaid, router)
+
+        if not uaid:
+            uaid = uuid.uuid4()
+            d = defer.execute(
+                router.register,
```

Comment:
We are inconsistent with how we're calling router.register (sometimes via named args, sometimes via ordered.) We had gone to named args prior because ordered had lead to some sleeper bugs. I would recommend we stay with that practice so that future coders don't hit the same problems.